src/cunumeric/isnan.h

```diff
@@ -0,0 +1,51 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include "legion.h"
+
+namespace cunumeric {
+
+template <legate::LegateTypeCode CODE>
+struct isnan {
```

Comment:
two comments: 1) I think this file belongs to the `unary` subdirectory; 2) you should use the following function templates instead:
```
template <typename T, std::enable_if_t<std::is_integral<T>::value>* = nullptr>
constexpr bool isnan(const T& x)
{
  return false;
}

template <typename T, std::enable_if_t<std::is_floating_point<T>::value>* = nullptr>
__CUDA_HD__ bool isnan(const T& x)
{
  return std::isnan(x);
}

template <typename T>
__CUDA_HD__ bool isnan(const complex<T>& x)
{
  return std::isnan(x.imag()) || std::isnan(x.real());
}

__CUDA_HD__ bool isnan(const __half& x)
{
  return isnan(x);
}
```

The reason that a functor is used in the unary op is that some unary operators need extra arguments and a functor makes it easier to pass them to those operators. A functor for a standalone `isnan` implementation only makes the caller code ugly (as you have to do `cunumeric::isnan()(argument)` instead of just `cunumeric::isnan(argument)`).